index.html

```diff
@@ -140,67 +166,250 @@ <h2>
+            It is an error to call the API from a worker that is not a service
+            worker.
+          </p>
+        </div>
         <p>
-          When the <code>clear()</code> function is called, set the application's
-          badge to <a>nothing</a>.
+          The steps to <dfn data-lt=
+          "determining the matching applications">determine the matching
+          applications</dfn> returns a sequence of <a data-cite=
+          "appmanifest#installable-web-applications">applications</a> or a
+          {{DOMException}}:
         </p>
+        <ol>
+          <li>If <a data-cite="html#current-settings-object">current settings
+          object</a> has a <a data-cite="html#responsible-document">responsible
+          document</a> <var>document</var>:
+            <ol>
+              <li>Let <var>apps</var> be the <a data-cite="infra#ordered-set">
+                set</a> of <a data-cite=
+                "appmanifest#installable-web-applications">installed web
+                applications</a> such that <var>document</var>'s <a data-cite=
+                "dom#concept-document-url">URL</a> is <a data-cite=
+                "appmanifest#dfn-within-scope-manifest">within scope</a> of the
+                application's manifest. (Order is not important.)
+              </li>
+              <li>
+                <a data-cite="infra#list-remove">Remove</a> all elements of
+                <var>apps</var> other than the one with the longest (i.e., most
+                specific) <a data-cite=
+                "appmanifest#dfn-navigation-scope">navigation scope</a>.
+              </li>
+              <li>Return <var>apps</var>.
+              </li>
+            </ol>
+          </li>
+          <li>Otherwise, if the <a data-cite=
+          "html#current-settings-object">current settings object</a> is a
+          <a data-cite="service-workers-1#dfn-service-worker-client">service
+          worker client</a> <var>worker</var>:
+            <ol>
+              <li>Return the <a data-cite="infra#ordered-set">set</a> of
+              <a data-cite="appmanifest#installable-web-applications">installed
+              web applications</a> whose <a data-cite=
+              "appmanifest#dfn-navigation-scope">navigation scope</a> is
+              <a data-cite="appmanifest#dfn-within-scope-manifest">within
+              scope</a> (as defined in [[AppManifest]]) of <var>worker</var>'s
+              <a data-cite=
+              "service-workers-1#dfn-containing-service-worker-registration">containing
+              service worker registration</a>'s <a data-cite=
+              "service-workers-1#dfn-scope-url">scope url</a>. (Order is not
+              important.)
+              </li>
+            </ol>
+          </li>
+          <li>Otherwise (this is a worker, but not a service worker), return a
+          {{NotSupportedError}}.
+          </li>
+        </ol>
       </section>
     </section>
+    <section class="informative">
```

Comment:
Let's make this "normative", even if there are no normative requirements here. 

```suggestion
    <section>
```

I'd also suggest we add the accessibility consideration. But we can do that as a different PR to close #24.

Reply:
Generally trying to move away from that "non-normative" trend specially for priv/sec sections. There are specs that do put normative requirements in place, for example (though it is amongst some non-normative sections):
https://www.w3.org/TR/payment-request/#exposing-user-information

Having said that, we can always flip this switch later if we do use some RFC2119 style requirements.